ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/common/FailureXAResource.java

```diff
@@ -9,146 +9,166 @@
+            }
+        }
+
+        // successful
+        this._committed = true;
+    }
+
+    public void end(Xid xid, int flags) throws XAException {
+        if (_locale == FailLocation.end) {
+            switch (_type) {
+                case normal:
+                    throw new XAException(XAException.XA_HEURRB);
+                case timeout:
+                    throw new XAException(XAException.XA_RBTIMEOUT);
+                case XA_RBCOMMFAIL:
+                    throw new XAException(XAException.XA_RBCOMMFAIL);
+            }
         }
     }
 
-    public void forget(Xid xid) throws XAException
-    {
+    public void forget(Xid xid) throws XAException {
     }
 
-    public int getTransactionTimeout() throws XAException
-    {
-	return 0;
+    public int getTransactionTimeout() throws XAException {
+        return 0;
     }
 
-    public boolean isSameRM(XAResource xares) throws XAException
-    {
-	return false;
+    public boolean isSameRM(XAResource xares) throws XAException {
+        return false;
     }
 
-    public int prepare(Xid xid) throws XAException
-    {
-        if ((_locale == FailLocation.prepare) || (_locale == FailLocation.prepare_and_rollback))
-        {
-            if (_type == FailType.message) {
-                XAException xae = new XAException(XAException.XA_RBROLLBACK);
-                xae.initCause(new Throwable("test message"));
-                throw xae;
-            } else if(_type == FailType.XA_HEURHAZ) {// XA spec invalid error code
-                throw new XAException(XAException.XA_HEURHAZ);
-            } else {
-                throw new XAException(XAException.XAER_INVAL);
+    public int prepare(Xid xid) throws XAException {
+        if ((_locale == FailLocation.prepare) || (_locale == FailLocation.prepare_and_rollback)) {
+            switch (_type) {
+                case message:
+                    XAException xae = new XAException(XAException.XA_RBROLLBACK);
+                    xae.initCause(new Throwable("test message"));
+                    throw xae;
+                case XA_RBINTEGRITY:
+                    throw new XAException(XAException.XA_RBINTEGRITY);
+                case XA_HEURHAZ: // XA spec invalid error code
+                    throw new XAException(XAException.XA_HEURHAZ);
+                default:
+                    throw new XAException(XAException.XAER_INVAL);
             }
         }
-        
+
         return XA_OK;
     }
 
-    public Xid[] recover(int flag) throws XAException
-    {
+    public Xid[] recover(int flag) throws XAException {
         return null;
     }
 
-    public void rollback(Xid xid) throws XAException
-    {
-        if ((_locale == FailLocation.rollback) || (_locale == FailLocation.prepare_and_rollback))
-        {
-            if (_type == FailType.normal)
-                throw new XAException(XAException.XA_HEURMIX);
-            
-            if (_type == FailType.heurcom)
```

Comment:
What prompted the change from the `if (_type...` to a `switch` style of comparison?

Reply:
I've moved this change into the STYLE commit. Thanks Tom